src/base/bittorrent/session.cpp

```diff
@@ -1244,65 +1243,84 @@ void Session::loadLTSettings(lt::settings_pack &settingsPack)
     // It will not take affect until the listen_interfaces settings is updated
     settingsPack.set_int(lt::settings_pack::listen_queue_size, socketBacklogSize());
 
-#ifdef Q_OS_WIN
-    QString chosenIP;
-#endif
     if (!m_listenInterfaceConfigured) {
         const int port = useRandomPort() ? 0 : this->port();
         if (port > 0)  // user specified port
             settingsPack.set_int(lt::settings_pack::max_retry_port_bind, 0);
 
-        for (const QString &ip : asConst(getListeningIPs())) {
-            if (ip.isEmpty()) {
-                const QString anyIP = QHostAddress(QHostAddress::AnyIPv4).toString();
-                const std::string endpoint = anyIP.toStdString() + ':' + std::to_string(port);
-                settingsPack.set_str(lt::settings_pack::listen_interfaces, endpoint);
-                LogMsg(tr("Trying to listen on IP: %1, port: %2"
-                        , "e.g: Trying to listen on IP: 192.168.0.1, port: 6881")
-                        .arg(anyIP, QString::number(port))
-                    , Log::INFO);
-                break;
-            }
+        QStringList endpoints;
+        QStringList outgoingInterfaces;
+        const QStringList listeningIPs = getListeningIPs();
+        const QString portString = ':' + QString::number(port);
 
-            lt::error_code ec;
-            const lt::address addr = lt::address::from_string(ip.toStdString(), ec);
-            if (!ec) {
-                const std::string endpoint = (addr.is_v6()
-                    ? ('[' + addr.to_string() + ']')
-                    : addr.to_string())
-                    + ':' + std::to_string(port);
-                settingsPack.set_str(lt::settings_pack::listen_interfaces, endpoint);
-                LogMsg(tr("Trying to listen on IP: %1, port: %2"
-                        , "e.g: Trying to listen on IP: 192.168.0.1, port: 6881")
-                        .arg(ip, QString::number(port))
-                    , Log::INFO);
+        const QString firstIP = listeningIPs[0];
+        if (firstIP.isEmpty()) {
+            endpoints << QHostAddress(QHostAddress::AnyIPv4).toString() + portString;
+            endpoints << '[' + QHostAddress(QHostAddress::AnyIPv6).toString() + ']' + portString;
+        }
+        else {
+            auto checkAndAddEndpoint = [&] (const QString &nextIP)
```

Comment:
maybe rename `nextIP` -> `IP`?

Reply:
Agree. `nextIP` is invalid name for this function since the function knows about one ip only. `ip` is better name. But, IIRC, it can contains not only IP... Maybe there can be more appropriate name?